File: jvm/scala/fizzy/src/main/scala/us/yellosoft/Fizzy.scala
package us.yellosoft.fizzy

object Fizzy {
    def fizzy : Seq[String] = (1 to 100).par.map { n =>
        (n % 3, n % 5) match {
            case (0, 0) => "FizzBuzz"
            case (0, _) => "Fizz"
            case (_, 0) => "Buzz"
            case _ => n.toString
        }
    }.seq

    def main(args : Array[String]) : Unit = fizzy.foreach(println)
}

File: system/java/fizzy/src/main/java/us/yellosoft/fizzy/Fizzy.java
package us.yellosoft.fizzy;

import java.util.stream.IntStream;

/** Example FizzBuzz implementation, parallelized */
public final class Fizzy {
    /** Utility class */
    private Fizzy() {}

    /** Modulo-based FizzBuzz implementation
     *
     * @param i Input number
     *
     * @return fizzbuzz(i)
     */
    public static String fizzbuzz(final int i) {
        final var m = i % 3 == 0;
        final var n = i % 5 == 0;

        if (m && n) {
            return "FizzBuzz";
        }

        if (m) {
            return "Fizz";
        }

        if (n) {
            return "Buzz";
        }

        return String.valueOf(i);
    }

    /** CLI entry point
     *  @param args CLI flags (unused)
     */
    public static void main(final String[] args) {
        final Iterable<String> strings = () -> IntStream
            .range(1, 101)
            .parallel()
            .mapToObj(Fizzy::fizzbuzz)
            .iterator();

        for (final var s: strings) {
            System.out.println(s);
        }
    }
}
